### PR TITLE
Revert SDK update

### PR DESCRIPTION
**Pull request by jkotalik**
Currently internal builds are failing due to mismatching msbuild versions.

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
   "sdk": {
-    "version": "5.0.100-rc.1.20379.10"
+    "version": "5.0.100-preview.7.20330.3"
```
Comment (dougbu) [inline global.json:3]: Suggest we use the public preview version, `5.0.100-preview.7.20366.6`. Hopefully this is a very short-lived change, so this is a nit.

**Review by dougbu — APPROVED** [review]
If the builds succeed, :shipit: 😺

**Comment by dougbu** [comment]
Looks like some of the work done in #24496 needs to be reverted to get tests working. For example,
``` text
ILlinker: error IL1005: Interop/Globalization::EnumCalendarInfo(method System.Void (System.Char,System.IntPtr),System.String,System.Globalization.CalendarId,System.Globalization.CalendarDataType,System.IntPtr): Error processing method 'System.Boolean Interop/Globalization::EnumCalendarInfo(method System.Void 
```
and, `dotnet publish` is failing frequently with a `NullReferenceException`.

**Comment by jkotalik** [comment]
I can try, but my gut says those failures are unrelated to anything that can be reverted in #24496

**Comment by jkotalik** [comment]
I don't think I'll be able to figure out why these tests are failing via reverting; I believe it's a new issue. Going to see if https://github.com/dotnet/aspnetcore/pull/24632 makes headwind before continuing to update this/ ask the right people to investigate.

**Comment by Pilchie** [comment]
:eyes:

**Comment by jkotalik** [comment]
Error that is being hit; it looks like we can't revert to an older SDK would be my naïve guess. 
```
[5.817s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:   Microsoft.AspNetCore.Server.Kestrel -> C:\Users\jukotali\code\aspnetcore2\artifacts\bin\Microsoft.AspNetCore.Server.Kestrel\Debug\net5.0\Microsoft.AspNetCore.Server.Kestrel.dll
[5.976s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:   BasicLinkedApp -> C:\Users\jukotali\code\aspnetcore2\src\Hosting\test\testassets\BasicLinkedApp\bin\Debug\net5.0\win-x64\BasicLinkedApp.dll
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:   Mono.Linker.LinkerFatalErrorException: ILlinker: error IL1005: Interop/Kernel32::EnumCalendarInfoExEx(method Interop/BOOL *(System.Char*,System.UInt32,System.IntPtr,System.Void*),System.String,System.UInt32,System.String,System.UInt32,System.Void*): Error processing method 'System.Boolean Interop/Kernel32::EnumCalendarInfoExEx(method Interop/BOOL *(System.Char*,System.UInt32,System.IntPtr,System.Void*),System.String,System.UInt32,System.String,System.UInt32,System.Void*)' in assembly 'System.Private.CoreLib.dll'
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:    ---> System.NullReferenceException: Object reference not set to an instance of an object.
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.ProcessInteropMethod(MethodDefinition method)
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method, DependencyInfo& reason)
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.ProcessQueue()
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      --- End of inner exception stack trace ---
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.ProcessQueue()
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.ProcessPrimaryQueue()
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.Process()
[11.603s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
[11.604s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
[11.604s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Pipeline.Process(LinkContext context)
[11.604s] [Microsoft.AspNetCore.Server.IntegrationTesting.SelfHostDeployer] [Information] dotnet-publish stdout:      at Mono.Linker.Driver.Run(ILogger customLogger)
```

**Comment by pranavkm** [comment]
That error was fixed in the linker in a p8 or newer SDK.

**Comment by jkotalik** [comment]
@pranavkm that means this PR will never work, correct?

**Comment by dougbu** [comment]
But, @pranavkm this PR failed validation when it just went back to the SDK we were successfully using. Admittedly that was before reverting some reactions to the new SDK.

Did we take a dependency on the new SDK somewhere after the update❔ If not, should `5.0.100-preview.7.20330.3` work w/ the changes @jkotalik has made so far❔

File: global.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
   "sdk": {
-    "version": "5.0.100-rc.1.20379.10"
+    "version": "5.0.100-preview.7.20366.6"
   },
   "tools": {
-    "dotnet": "5.0.100-rc.1.20379.10",
+    "dotnet": "5.0.100-preview.7.20366.6",
```
Comment (dougbu) [inline global.json:6]: Put another way, my low-pri suggestion here might have been **Just Wrong:tm:** Oopsies

**Comment by jkotalik** [comment]
I'll try the right sdk then...

**Comment by jkotalik** [comment]
@dougbu that didn't work ☹️

**Pull request closed without merging**